src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java

```diff
@@ -522,20 +704,32 @@ protected OriginEntryFull buildCorpCardPersonalExpenseOriginEntry(
         return originEntry;
     }
 
+    protected OriginEntryFull buildAtmFeeDebitOriginEntry(ConcurStandardAccountingExtractDetailLine detailLine, KualiDecimal amount) {
+        OriginEntryFull originEntry = new OriginEntryFull();

```

Comment:
there are a few build origin entry functions here.  Can these be refactored to share functionality?

Reply:
Most of the shared OriginEntryFull setup is in the "configureOriginEntryGeneratedFromLine" method; the other accounting data setup may not be easy to consolidate because of the varying requirements for the entries being generated. However, after looking further, I think the "new OriginEntryFull()" call can also be consolidated into that helper method, so I'll try to refactor that accordingly.